streams/src/main/java/org/apache/kafka/streams/processor/DefaultPartitionGrouper.java

```diff
@@ -61,7 +61,7 @@ protected int maxNumPartitions(Cluster metadata, Set<String> topics) {
             List<PartitionInfo> infos = metadata.partitionsForTopic(topic);
 
             if (infos == null)
-                throw new KafkaException("topic not found :" + topic);
+                throw new KafkaException("Topic not found during partition assignment: " + topic);
```

Comment:
My reasoning in Connect was that we wanted to be able to isolate things that Connect was aware of and was truly unable to handle from other stuff that could bubble through. What this *should* mean (assuming Connect handles things properly...) is that you can just catch ConnectException, even in embedded mode, and anything else is pretty much unrecoverable anyway (so catch RuntimeException and log it, but you can't really do anything about it). In Connect this is kind of important because you might get random exceptions from connectors (and the libraries they are using) so I wanted something that would be as predictable and isolated as possible.

I haven't thought through the implications for KStreams though. Not sure you necessarily need anything quite as strict, and as long as the expectations about source/sink topics possibly throwing KafkaException is clear, maybe that's fine. Just wanted to make sure there was a consistent policy.